contract_compassion/model/contracts.py

```diff
@@ -0,0 +1,646 @@
+        res = dict()
+        for contract in self.browse(cr, uid, ids, context):
+            count = self.pool.get('account.banking.mandate').search(cr, uid, [
+                ('partner_id', '=', contract.partner_id.id),
+                ('state', '=', 'valid')], count=True, context=context)
+            res[contract.id] = bool(count)
+        return res
+
+    def _name_get(self, cr, uid, ids, field_name, args, context=None):
+        return {c[0]: c[1] for c in self.name_get(cr, uid, ids, context)}
+
+    ###########################
+    #        New Fields       #
+    ###########################
+    _columns = {
+        'child_id': fields.many2one(
+            'compassion.child', _('Sponsored child'), readonly=True,
+            states={'draft': [('readonly', False)]}, ondelete='restrict'),
+        'child_name': fields.related(
+            'child_id', 'name', string=_('Sponsored child name'),
+            readonly=True, type='char'),
+        'child_code': fields.related(
+            'child_id', 'code', string=_('Sponsored child code'),
+            readonly=True, type='char'),
+        'partner_codega': fields.related(
+            'correspondant_id', 'ref', string=_('Partner ref'), readonly=True,
+            type='char'),
+        'correspondant_id': fields.many2one(
+            'res.partner', _('Correspondant'), required=True, readonly=True,
+            states={'draft': [('readonly', False)]}),
+        'activation_date': fields.date(
+            _('Activation date'), readonly=True),
+        # Add a waiting and waiting mandate states
+        'state': fields.selection([
+            ('draft', _('Draft')),
+            ('mandate', _('Waiting Mandate')),
+            ('waiting', _('Waiting Payment')),
+            ('active', _('Active')),
+            ('terminated', _('Terminated')),
+            ('cancelled', _('Cancelled'))], _('Status'), select=True,
+            readonly=True, track_visibility='onchange',
+            help=_(" * The 'Draft' status is used when a user is encoding a "
+                   "new and unconfirmed Contract.\n"
+                   " * The 'Waiting' status is used when the Contract is "
+                   " confirmed but the partner has not yet paid.\n"
+                   " * The 'Active' status is used when the contract is "
+                   "confirmed and until it's terminated.\n"
+                   " * The 'Terminated' status is used when a contract is no "
+                   "longer active.\n"
+                   " * The 'Cancelled' status is used when a contract was "
+                   "never paid.")),
+        'is_active': fields.function(
+            _active, string='Contract Active', type='boolean',
+            store={
+                'account.invoice': (_get_contract_from_invoice, ['state'], 50)
+            },
+            help="It indicates that the first invoice has been paid and the "
+                 "contract was activated."),
+        'fully_managed': fields.function(

```

Comment:
All this functionality can be moved in sponsorship_compassion. A regular contract is always fully_managed (no correspondant).